mongo/integration/unified/client_entity_test.go

```diff
@@ -49,10 +49,13 @@ func NewClientEntity(ctx context.Context, entityOptions *EntityOptions) (*Client
 			return nil, fmt.Errorf("error parsing URI options: %v", err)
 		}
 	}
-	// UseMultipleMongoses only requires options to be set if we're connected to a sharded cluster and it's set to
-	// false. If it's unset or true, we make no changes because the cluster URI already includes all mongos nodes.
-	if mtest.ClusterTopologyKind() == mtest.Sharded && !entityOptions.UseMultipleMongoses {
-		clientOpts.SetHosts(mtest.ClusterConnString().Hosts[:1])
+	// UseMultipleMongoses is only relevant if we're connected to a sharded cluster. Options changes and validation are
+	// only required if the option is explicitly set. If it's unset, we make no changes because the cluster URI already
+	// includes all nodes and we don't enforce any limits on the number of nodes.
+	if mtest.ClusterTopologyKind() == mtest.Sharded && entityOptions.UseMultipleMongoses != nil {
```

Comment:
Also check if `ShardedReplicaSet`?

Reply:
`ClusterTopologyKind` returns `Sharded` for both sharded and sharded-replicaset. The sharded-replicaset constraint is checked by keeping an additional boolean. See https://github.com/mongodb/mongo-go-driver/blob/171476d0dbdc9d657e01c5acaeb397052c818730/mongo/integration/mtest/setup.go#L122.

Reply:
Ah that's right...